assets/src/util/assignment.js

```diff
@@ -8,39 +8,60 @@ const clearGoals = assignments => assignments
     return a
   })
 
-const setAssigmentGoalInputState = (key, assignments, inputFocus) => {
-  return [
-    ...assignments.slice(0, key),
-    {
-      ...assignments[key],
-      inputFocus,
-      inputBlur: !inputFocus
-    },
-    ...assignments.slice(key + 1)
-  ]
+const setAssigmentGoalInputState = (assignmentId, assignments, inputFocus) => {
+  const assignment = assignments.filter(a => a.id === assignmentId)
+  if (assignment.length !== 1) {
+    console.log('Error finding unique assignment id')
```

Comment:
This is one of those "should never happen" things.  To better handle that would need to change the way the function is used.  Same with the other 2 of this ilk.

Reply:
I don't think if we want to put console.log in JS code since it might appear browser for user, this case is a rare case.
But I think this can be handles like this https://stackoverflow.com/questions/41040266/remove-console-logs-with-webpack-uglify/41041580#41041580.

Reply:
but since you are trying to find an assignment id from a list if you find 2 can you check if both are same and remove one. can that be handled like that?

Reply:
@ssciolla @jonespm what are your thoughts on having console.log in JS code?

Reply:
Hm, yeah, these cases can be tricky. In terms of logging in production, I've seen some people do it. Not sure if it's necessary here. What happens in the UI if this occurs? In this case, `console.error` would be more appropriate.

Reply:
In the UI it would simply not apply the filter b/c its in a bad/unexpected state.. trying to find an assignment with the provided id (that it got from the client side - so it should exist) and it can't or it finds it more than once.  Neither of those situations should happen.  console.error would make more sense.. i left it in as someway that someone would be able to provide some feedback if they observed strange behavior, but it would obviously rely on that being manually provided to someone supporting the issue..